ClassCastException: VAlarm cannot be cast to Observance when VTIMEZONE contains a VALARM

(ical4j 3.2.17, the VCALENDAR was received over CalDAV by DAVx5)

When a VTIMEZONE definition contains a VALARM (which is obviously not allowed), ical4j currently throws an Exception and stops parsing instead of ignoring the invalid VALARM:

<details><summary>Test</summary>
<pre lang="kotlin">
    @Test
    fun testTzDefWithVAlarm() {
        val builder = CalendarBuilder()
        val cal = builder.build(StringReader("BEGIN:VCALENDAR\n" +
                "VERSION:2.0\n" +
                "BEGIN:VTIMEZONE\n" +
                "TZID:Europe/Zurich\n" +
                "BEGIN:DAYLIGHT\n" +
                "TZOFFSETFROM:+0100\n" +
                "TZOFFSETTO:+0200\n" +
                "TZNAME:CEST\n" +
                "DTSTART:19810329T020000\n" +
                "RRULE:FREQ=YEARLY;BYMONTH=3;BYDAY=-1SU\n" +
                "END:DAYLIGHT\n" +
                "BEGIN:STANDARD\n" +
                "TZOFFSETFROM:+0200\n" +
                "TZOFFSETTO:+0100\n" +
                "TZNAME:CET\n" +
                "DTSTART:19961027T030000\n" +
                "RRULE:FREQ=YEARLY;BYMONTH=10;BYDAY=-1SU\n" +
                "END:STANDARD\n" +
                "BEGIN:VALARM\n" +
                "ACTION:DISPLAY\n" +
                "DESCRIPTION:Reminder\n" +
                "TRIGGER:-PT15M\n" +
                "END:VALARM\n" +
                "END:VTIMEZONE\n" +
                "BEGIN:VEVENT\n" +
                "SUMMARY:Test\n" +
                "DTSTART;TZID=Europe/Zurich:20240314T111500\n" +
                "DTEND;TZID=Europe/Zurich:20240314T114500\n" +
                "END:VEVENT\n" +
                "END:VCALENDAR\n"))
    }
</pre>
</details> 

→
```
Caused by: java.lang.ClassCastException: net.fortuna.ical4j.model.component.VAlarm cannot be cast to net.fortuna.ical4j.model.component.Observance
	at net.fortuna.ical4j.model.component.VTimeZone.getApplicableObservance(VTimeZone.java:21)
	at net.fortuna.ical4j.model.TimeZone.inDaylightTime(TimeZone.java:8)
	at java.util.TimeZone.getOffsets(TimeZone.java:247)
	at java.util.GregorianCalendar.adjustForZoneAndDaylightSavingsTime(GregorianCalendar.java:2916)
	at java.util.GregorianCalendar.computeTime(GregorianCalendar.java:2807)
	at java.util.Calendar.updateTime(Calendar.java:3403)
	at java.util.Calendar.getTimeInMillis(Calendar.java:1761)
	at java.util.Calendar.getTime(Calendar.java:1734)
	at java.text.SimpleDateFormat.parseInternal(SimpleDateFormat.java:1831)
	at java.text.SimpleDateFormat.parse(SimpleDateFormat.java:1725)
	at java.text.DateFormat.parse(DateFormat.java:360)
	at net.fortuna.ical4j.model.DateTime.setTime(DateTime.java:2)
	at net.fortuna.ical4j.model.DateTime.<init>(DateTime.java:22)
	at net.fortuna.ical4j.model.property.DateProperty.setValue(DateProperty.java:39)
	at net.fortuna.ical4j.data.DefaultContentHandler.resolveTimezones(DefaultContentHandler.java:80)
	at net.fortuna.ical4j.data.DefaultContentHandler.endCalendar(DefaultContentHandler.java:1)
	at net.fortuna.ical4j.data.CalendarParserImpl.parseCalendar(CalendarParserImpl.java:50)
	at net.fortuna.ical4j.data.CalendarParserImpl.parseCalendarList(CalendarParserImpl.java:15)
	at net.fortuna.ical4j.data.CalendarParserImpl.parse(CalendarParserImpl.java:13)
```

This is because `VTimeZone.getApplicableObservance` calls `getObservances()`, which takes the `components` of the `VTIMEZONE` and assumes that all components are Observances:

https://github.com/ical4j/ical4j/blob/32f7de231961a657162293f131942aaeca29b457/src/main/java/net/fortuna/ical4j/model/component/VTimeZone.java#L181-L183

In the case of this invaild VCALENDAR, this is not the case.

Suggestion: only get the components that are actually observances in `getObservances()`.

--- a/src/main/java/net/fortuna/ical4j/model/component/VTimeZone.java
+++ b/src/main/java/net/fortuna/ical4j/model/component/VTimeZone.java
@@ -179,12 +179,14 @@ public class VTimeZone extends CalendarComponent implements ComponentContainer<O
      * @return Returns the types.
      */
     public final ComponentList<Observance> getObservances() {
-        return (ComponentList<Observance>) components;
+        ComponentList<? extends Component> list = (ComponentList<? extends Component>) components.clone();
+        list.removeIf(s -> !(s instanceof Observance));
+        return (ComponentList<Observance>) list;
     }
 
     @Override
     public ComponentList<Observance> getComponents() {
-        return (ComponentList<Observance>) components;
+        return getObservances();
     }
 
     /**

